test/lifecycle_module_test.dart

```diff
@@ -149,6 +149,7 @@ class TestLifecycleModule extends LifecycleModule {
   @override
   @protected
   Future<Null> onWillUnloadChildModule(LifecycleModule module) async {
+    await new Future.value(null);
```

Comment:
This spot is a little more concerning because this may mean that it's possible for other consumers that are overriding one of these lifecycle methods may get a change in how lifecycle events are ordered due to sync-async. I think we can go ahead with this PR as is, but be aware that there may be a recommended migration step of putting a line like this in each overriden module lifecycle method.

Reply:
Without this await here .. the order that events are added to the tracking list of events is different than what the tests expect the order to be. This looks like:

```
00:05 +216 -1: [VM] test/lifecycle_module_test.dart: LifecycleModule with globalTracer with children unload should wait for in-progress child module loads  [E]
  Expected: [
              'onWillLoadChildModule',
              'willLoadChildModule',
              'onShouldUnload',
              'willUnload',
              'onDidLoadChildModule',
              'didLoadChildModule',
              'onWillUnloadChildModule',
              'willUnloadChildModule',
              'onDidUnloadChildModule',
              'didUnloadChildModule',
              'onUnload',
              'didUnload',
              'onDispose'
            ]
    Actual: [
              'onWillLoadChildModule',
              'willLoadChildModule',
              'onShouldUnload',
              'willUnload',
              'onDidLoadChildModule',
              'onWillUnloadChildModule',
              'didLoadChildModule',
              'willUnloadChildModule',
              'onDidUnloadChildModule',
              'didUnloadChildModule',
              'onUnload',
              'didUnload',
              'onDispose'
            ]
     Which: was 'onWillUnloadChildModule' instead of 'didLoadChildModule' at location [5]

  package:test                             expect
  test/lifecycle_module_test.dart 2236:11  runTests.<fn>.<fn>.<fn>.<fn>
```